From 18ddd5df2c701eff3e450f4fd8f45c1254cfbfa5 Mon Sep 17 00:00:00 2001
From: Rabeeh Khoury <rabeeh@solid-run.com>
Date: Sat, 3 May 2014 15:58:44 +0300
Subject: [PATCH] Some hacks

1. Hacks to get tinyalsa forced to choose card 1 (HDMI sink)
2. Use front LED instead of backlight
3. LK 3.10.30 doesn't support boostpulse.

Those are hacks just to silence logcat and get things going. They should
be revisited (espcially the tinyalsa) and worked out.

Signed-off-by: Rabeeh Khoury <rabeeh@solid-run.com>
---
 alsa/tinyalsa_hal.c |    6 +++++-
 lights/fsl_lights.c |    4 ++--
 mx6/power/power.cpp |    2 +-
 3 files changed, 8 insertions(+), 4 deletions(-)

diff --git a/alsa/tinyalsa_hal.c b/alsa/tinyalsa_hal.c
index 5084138..55dbdb2 100644
--- a/alsa/tinyalsa_hal.c
+++ b/alsa/tinyalsa_hal.c
@@ -16,7 +16,7 @@
 /* Copyright (C) 2012-2013 Freescale Semiconductor, Inc. */
 
 #define LOG_TAG "audio_hw_primary"
-//#define LOG_NDEBUG 0
+#define LOG_NDEBUG 0
 
 #include <errno.h>
 #include <pthread.h>
@@ -487,9 +487,12 @@ static int get_card_for_device(struct imx_audio_device *adev, int device, unsign
 {
     int i;
     int card = -1;
+    ALOGV("get_card_from_device flag = %d (PCM_OUT = %d) MAX_AUDIO_CARD_NUM = %d\n", flag,PCM_OUT,MAX_AUDIO_CARD_NUM);
 
     if (flag == PCM_OUT ) {
         for(i = 0; i < MAX_AUDIO_CARD_NUM; i++) {
+	    ALOGV("get_card_from_device %d : 0x%x 0x%x card = %d\n", i, adev->card_list[i]->supported_out_devices, device,adev->card_list[i]->card);
+
             if(adev->card_list[i]->supported_out_devices & device) {
                   card = adev->card_list[i]->card;
                   break;
@@ -503,6 +506,7 @@ static int get_card_for_device(struct imx_audio_device *adev, int device, unsign
             }
         }
     }
+    card = 1; // Rabeeh - hack for LK 3.10.30
     return card;
 }
 /* must be called with hw device and output stream mutexes locked */
diff --git a/lights/fsl_lights.c b/lights/fsl_lights.c
index 2157e0c..ff8e3ff 100644
--- a/lights/fsl_lights.c
+++ b/lights/fsl_lights.c
@@ -26,8 +26,8 @@
 #include <cutils/properties.h>
 
 #define MAX_BRIGHTNESS 255
-#define DEF_BACKLIGHT_DEV "pwm-backlight"
-#define DEF_BACKLIGHT_PATH "/sys/class/backlight/"
+#define DEF_BACKLIGHT_DEV "imx6\:red\:front"
+#define DEF_BACKLIGHT_PATH "/sys/class/leds/"
 
 /*****************************************************************************/
 struct lights_module_t {
diff --git a/mx6/power/power.cpp b/mx6/power/power.cpp
index 5db7217..68e36d7 100644
--- a/mx6/power/power.cpp
+++ b/mx6/power/power.cpp
@@ -109,7 +109,7 @@ static void fsl_power_hint(struct power_module *module, power_hint_t hint,
     case POWER_HINT_VSYNC:
         break;
     case POWER_HINT_INTERACTION:
-        sysfs_write(BOOSTPULSE_PATH, "1");
+//        sysfs_write(BOOSTPULSE_PATH, "1");
         break;
     default:
         break;
-- 
1.7.9.5

